File: server/app/models/auth/signup_request.py
from typing import List
from pydantic import BaseModel
from app.db.enums import Gender, AvailableCourse, AvailableDialect


class SignupRequest(BaseModel):
    email: str
    password: str
    username: str
    first_name: str
    last_name: str | None = None
    gender: Gender
    current_course: AvailableCourse | None = None
    current_dialect: AvailableDialect | None = None
    languages_learning: List[str] = []
    languages_learned: List[str] = []

File: client/types/request_models/SignupRequest.ts
import { AvailableCourse, AvailableDialect, Gender } from "../enums"

export interface SignupRequest {
    email: string
    password: string
    username: string
    first_name: string
    last_name: string | null
    gender: Gender
    current_course: AvailableCourse | null
    current_dialect: AvailableDialect | null
    languages_learning: string[]
    languages_learned: string[]
}